tests/lib/rules/no-unused-state.js

```diff
@@ -636,6 +636,64 @@ eslintTester.run('no-unused-state', rule, {
       }
       `,
       parser: 'babel-eslint'
+    }, {
+      code: `
+      var Foo = createReactClass({
+        getInitialState: function() {
+          return { initial: 'foo' };
+        },
+        handleChange: function() {
+          this.setState(state => ({
+            current: state.initial
+          }));
+        },
+        render() {
+          const { current } = this.state;
+          return <div>{current}</div>
+        }
+      });
+      `
+    }, {
+      code: `
+      var Foo = createReactClass({
+        getInitialState: function() {
+          return { initial: 'foo' };
+        },
+        handleChange: function() {
+          this.setState((state, props) => ({
+            current: state.initial
+          }));
+        },
+        render() {
+          const { current } = this.state;
+          return <div>{current}</div>
+        }
+      });
+      `
+    }, {
+      // Don't error out
+      code: `
+      class Foo extends Component {
+        handleChange = function() {
```

Comment:
This and the next test case were what I could find that would reproduce the error. It seems like weird usages, but it won't error out.